rename DoubleDataPoint to NumberDataPoint

codeboten opened the pull request:
**Description:** 
Rename DoubleDataPoint to NumebrDataPoint. Contrib repo update for https://github.com/open-telemetry/opentelemetry-collector/pull/3633

**Link to tracking Issue:** Part of https://github.com/open-telemetry/opentelemetry-collector/issues/3534

Commit:
rename DoubleDataPoint to NumberDataPoint

codeboten commented:
I see one of the tests failed, not immediately clear to me if it's related to this change or not
```
     integration_test.go:219: 
        	Error Trace:	integration_test.go:219
        	            				asm_amd64.s:1371
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 4
        	Test:       	TestJMXIntegration/TestJMXReceiverHappyPath
    integration_test.go:176: 
        	Error Trace:	integration_test.go:176
        	Error:      	Condition never satisfied
        	Test:       	TestJMXIntegration/TestJMXReceiverHappyPath
        	Messages:   	metrics not collected: 
```

codeboten commented:
> I see one of the tests failed, not immediately clear to me if it's related to this change or not
> 
> ```
>      integration_test.go:219: 
>         	Error Trace:	integration_test.go:219
>         	            				asm_amd64.s:1371
>         	Error:      	Not equal: 
>         	            	expected: 3
>         	            	actual  : 4
>         	Test:       	TestJMXIntegration/TestJMXReceiverHappyPath
>     integration_test.go:176: 
>         	Error Trace:	integration_test.go:176
>         	Error:      	Condition never satisfied
>         	Test:       	TestJMXIntegration/TestJMXReceiverHappyPath
>         	Messages:   	metrics not collected: 
> ```

Turns out the failure was caused by the change to translate `IntSum` to `Sum` https://github.com/open-telemetry/opentelemetry-collector/pull/3621

bogdandrutu commented on exporter/awscloudwatchlogsexporter/go.mod:
```diff
@@ -9,3 +9,7 @@ require (
 	go.uber.org/zap v1.18.1
 )
+
+replace go.opentelemetry.io/collector => go.opentelemetry.io/collector v0.29.1-0.20210716020257-4d8e3082465d
+
+replace go.opentelemetry.io/collector/model => go.opentelemetry.io/collector/model v0.0.0-20210716020257-4d8e3082465d
```
This is confusing, why these changes?

bogdandrutu replied:
Why not 0.30.0 something?

codeboten replied:
this is a good question. `go get` returned an errors about the versions not existing when I tried w/ 0.30.1

codeboten replied:
This is the versions I ended up with from using main

```
go get go.opentelemetry.io/collector@main
go get: downgraded go.opentelemetry.io/collector v0.30.0 => v0.29.1-0.20210716213518-a5a838ed3569

go get go.opentelemetry.io/collector/model@main
go get: downgraded go.opentelemetry.io/collector/model v0.30.0 => v0.0.0-20210716213518-a5a838ed3569
```

bogdandrutu commented:
Please fix the lint errors

codeboten commented:
> Please fix the lint errors

Addressed the errors, I reverted the `prom-counter` example back to using v0.20.0 of the otel go SDK, will update that in a separate PR.

codeboten commented:
Separate PR for the prom counter here https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/4261

Commit:
make update-otel

Commit:
Merge remote-tracking branch 'origin/main' into codeboten/rename-double-data-point

Commit:
revert change to tracegen/go.sum

Commit:
Merge remote-tracking branch 'origin/main' into codeboten/rename-double-data-point

Commit:
mod tidy
